arkouda/categorical.py

```diff
@@ -529,3 +534,197 @@ def concatenate(self, others : List[Categorical], ordered : bool=True) -> Catego
+        if components:
+            for k, v in components:
+                self.logger.debug(f"Un-registering Categorical attribute {k} with name {self.name}.{k}")
+                v.unregister()
+        self.name = None  # Clear our internal Categorical object name
+
+    def is_registered(self) -> np.bool_:
+        """
+         Return True iff the object is contained in the registry
+
+        Returns
+        -------
+        numpy.bool
+            Indicates if the object is contained in the registry
+
+        Raises
+        ------
+        RegistrationError
+            Raised if there's a server-side error or a mis-match of registered pieces
+
+        See Also
+        --------
+        register, attach, unregister, unregister_categorical_by_name
+
+        Notes
+        -----
+        Objects registered with the server are immune to deletion until
+        they are unregistered.
+        """
+        reg:bool
+        components = self._get_vars_with_register()
+        if components:
+            registered = [k for k, v in components if v.is_registered()]
+            if registered and len(registered) == len(components):  # everything is registered
+                reg = True
+            elif not registered:  # nothing is registered
+                reg = False
+            else:  # Some are registered but not all
+                raise RegistrationError(f"Not all registerable components of Categorical {self.name} are registered.")
+        else:
+            reg = False
+
+        return np.bool_(reg)  # Using numpy for consistency with other implementations
+
+    @staticmethod
+    @typechecked
+    def attach(user_defined_name:str) -> Categorical:
+        """
+        Function to return a Categorical object attached to the registered name in the
+        arkouda server which was registered using register()
+
+        Parameters
+        ----------
+        user_defined_name : str
+            user defined name which Categorical object was registered under
+
+        Returns
+        -------
+        Categorical
+               The Categorical object created by re-attaching to the corresponding server components
+
+       Raises
+       ------
+       TypeError
+            if user_defined_name is not a string
+
+        See Also
+        --------
+        register, is_registered, unregister, unregister_categorical_by_name
+        """
+        parts = {}  # Build dict of registered components by invoking their corresponding Class.attach functions
+        for k,v in Categorical.RegisterableComponents.items():
+            parts[k] = v.attach(f"{user_defined_name}.{k}")
+        # Now call constructor with kwargs and hand it the unpacked dictionary object
+        c = Categorical(None, **parts)
+        c.name = user_defined_name # Update our name
+        return c
+
+    @staticmethod
+    @typechecked
+    def unregister_categorical_by_name(user_defined_name:str) -> None:
```

Comment:
@glitch should we change the `pdarray` interface to match this?

Reply:
This is a good question,  right now `pdarrayclass.unregister_pdarray(...) ` takes either a String name or a `pdarray` object itself as an argument and currently the instance method `pda_obj.unregister()` delegates to the static method, so there would be some wiring changes.

I didn't want to run too far afield in this PR but I'm happy to make the change.  If you're only thinking of the naming for consistency I could do that instead.  Let me know what you think.

Reply:
I think we should maintain consistency in the interface.